OHIF/Viewers
# fix: get direct url pixel data should be optional for video

Refactor getDirectURL into three documented and tested functions. Also, this change allows building /rendered URLs for video instances that does not have pixel data present.








### Context

The get direct URL was a multi-purpose util function. This PR splits it into three independent functions that get called within to get the direct url.



### Changes & Results

Now getDirectURL calls getBulkDataValue and createRenderedRetrieve as fallbacks.



### Testing

- Load a video study using json loader (with the URL attribute present) and observe that it will be used as direct url and later as video src for video modality
- Load a study with DirectRetrieveURL and observe that it will be used as a direct URL
- Load a study with InlineBinary and observe that it will be used as a direct URL
- Load a study with either 'PixelData' or 'EncapsulatedDocument' and observe that /rendered URL is built and returned as getDirectURL
- Load a study with BulkDataURI and observe that it will be used as direct URL with the correct accept query param

(!singlepart || (singlepart !== true && singlepart.indexOf(fetchPart) === -1)



### Checklist

#### PR



  semantic-release format and guidelines.

#### Code

  etc.)

#### Public Documentation Updates



N/A

#### Tested Environment

- [x] "OS: 14.4.1 (23E224) 
- [x] "Node version: 16.14 
- [x] "Browser: Chrome Version 123.0.6312.59 (Official Build) (x86_64)
  


[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch
